File: Sokoban_2._0_Console/Sokoban_2._0_Console/Controls/Direction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sokoban_2._0_Console.Controls
{
    public class Direction
    {
        public enum dir {
            // The 4 Direction  we can choose
            UP, DOWN, LEFT, RIGHT
        }
        public static Direction UP => new Direction(dir.UP,dir.DOWN);
        public static Direction DOWN => new Direction(dir.DOWN, dir.UP);
        public static Direction RIGHT => new Direction(dir.RIGHT, dir.LEFT);
        public static Direction LEFT => new Direction(dir.LEFT, dir.RIGHT);

        public readonly dir direction;
        public readonly dir opposite;
        Direction(dir _direction, dir _opposite){ direction=_direction; opposite = _opposite; }

        // A function that returns the opposite for a certain direction
        public Direction GetReverse()
        {
            return new Direction(opposite,direction);
        }

    }
}

File: sokoban/src/hu/bme/sch/sokoban/game/commons/Direction.java
package hu.bme.sch.sokoban.game.commons;

public enum Direction {
    // The 4 Direction  we can choose
    UP, DOWN, LEFT, RIGHT;

    // Each direction has an opposite direction, these opposites are declared here
    private Direction opposite;
    static {
        UP.opposite = DOWN;
        DOWN.opposite = UP;
        RIGHT.opposite = LEFT;
        LEFT.opposite = RIGHT;
    }

    // A function that returns the opposite for a certain direction
    public Direction getReverse() {
        return opposite;
    }
}